### DISCUSSION TITLE
Final Project Idea Discussion

### DISCUSSION THREAD

**Discussion by eculler** [earthlab-education/earth-analytics-02-python-2024#1] [Projects]
A reminder @earthlab-education/ea-students — Put your ideas and interests here, and respond to each other! I’ll continue to jump in and add projects that have been proposed to me.

**Comment by annettebalmes** [comment]
Idea #1: I work for the geomagnetism group at CIRES at CU-Boulder. Our group works on [lots of things](https://geomag.colorado.edu/), etc. but we also help process magnetic data for various groups that ask for our help, mostly for my work of [implementing magnetic navigation ](https://geomag.colorado.edu/annette-balmes.html), which requires separating out the crustal magnetic field from the data.

That last task takes up a lot of my time because we only use a program called Oasis Montaj really to help process code and it's a really intensive process that requires constant visualizing and intervention at every step. Our group has barely created our own code/programs to help automate this task because we have to work on so many other things. I have begun, but my code is really written in a way that only I can understand and for specific cases and needs to be reworked immensely.

I want to write code to essentially create, ideally an interactive program, to do simple magnetic processing tasks. Nothing too crazy, but I theoretically want anyone to be able to run this program, not just those with previous magnetic processing/geophysics backgrounds. This would include something along the lines of:

- Organizing the data, marine or aeromag
- Calculating the crustal magnetic field and leveling the data
- Calculating uncertainty of the data
- Interpolating the data and creating a map

Every step requires visualization of the data and conducting and analyzing results to ensure the results are correct. There may also be code to write that will help to remove some of those steps and further "automate" the process.

Potential Issues:
* Scale- I'm not sure how likely it is that I could finish this by end of summer, something this coding intensive is really hard to time. I'm sure I can dedicate some of my actual work hours to this, but I also have a lot of deadlines coming up for my actual work.
* Grant Funded- I don't think the people who help fund me will want me releasing my code, I brought this up and they said I can always get help from mentors as it's not confidential, but they don't want me sharing unnecessarily so I definitely would have to do it by myself, maybe I don't want to risk it.
* Does this really fall within the scope of this class?

**Reply by eculler** [reply]
@annettebalmes this definitely represents a methodological technical challenge in Earth Data Science. You’re right that this is a lot. However, organizing the data plus one of the calculation steps would probably be feasible.

You can use a private repository and grant access as needed for reviews if you would like to pursue this.

**Comment by annettebalmes** [comment]
Idea #2: Outside of my field, I am also really interested in [Remote Sensing Image Classification](https://gisgeography.com/image-classification-techniques-remote-sensing/), maybe comparing the different techniques like unsupervised, supervised, and object-based, maybe training a neural network or something like that. 

Idea #3: Predicting Solar Panel Output using Satellite Imagery/Historical Weather. Interesting use case would be on my house and I could predict if I should get solar panels or not haha.

**Reply by edchanonline** [reply]
Yes, I would be interested in something along the line of *Remote Sensing Objects Detection + Deep Learning*. Perhaps putting together a dataset is also tangentially related? I was reading this at https://www.mdpi.com/2072-4292/13/18/3690 (SAR Ship Detection Dataset) but I would be interested to find out if there are similar researches being done here in the States. 

Umbra has been sharing one beautiful SAR image per day and is taking suggestions for any location on Earth. SAR images might be interested to look into.

**Reply by edchanonline** [reply]
Speaking of learning models, to add, I was loosely reading a book _Artificial Intelligence in Earth Science by Ziheng Sun, Nicoleta Cristea, Pablo Rivas_ which could serve us well in terms of refining additional ideas for this theme.

**Reply by eculler** [reply]
@annettebalmes @eggvoice I would like to run a project on rapid large-scale landslide identification using SAR amplitude data from Sentinel-1 (and possible other low-latency products, although that may not be in scope).  I have had students work on this in the past using a Bayesian change-detection algorithm, and would like to focus on scaling up, using some NSF cyberinfrastructure. I’m also hoping to recruit a big data pro to co-mentor. So that sounds like potentially a good fit! I’m also hoping to take a look at the new AR (analysis-ready) disturbances product from NASA derived from Sentinel-1 and multispectral data. If nothing else, it could narrow down the spatial scale and we could work on categorizing the type of disturbance using object recognition.

We also may have a project using hyperspectral imagery, which would be something more like carbon accounting, categorizing land cover classes, or mapping mangrove forests. That project is still pretty open, so if it’s intriguing let me know what about it you’re interested in diving into.

**Reply by edchanonline** [reply]
This honestly sounds amazing. I am just a little hesitant given I am looking for opportunities to work with a really great project partner/sponsor.

**Reply by misterskye** [reply]
I would definitely be interested...With that said, I know there are a few resources out there that are doing a pretty good job already:  HazMapper:  https://hazmapper.org/     and
David Milledge's Automated Landslide Detection Index:  https://github.com/DavidMilledge/ALDI/releases/tag/v1.1.0

I have been thinking about trying to automate where to locate landslide monitoring instruments.  Something that would ingest land use, slope, historical meteorological data, and others to determine approximate locations of sites that could help monitor landslide danger.

**Comment by edchanonline** [comment]
**Python Package Development**

I am interested in creating utilities to interact with different APIs. `earthpy` was mentioned. There are also newer platforms like SkyFi. I am already working on a side project where I am creating a simple Python Package that would help interact with SkyFi APIs to create timelapses of Satellite images. I have made a lot of progress already so probably need to expand this idea or find similar ideas.

**Reply by eculler** [reply]
@eggvoice you might want to take a look at [streamlit and Google Earth Engine](https://blog.streamlit.io/creating-satellite-timelapse-with-streamlit-and-earth-engine/) if you haven’t already.

I’d definitely be interested in publishing any API utilities you build in `earthpy`. We can also take a look at other packages (pyOpenSci and Pangeo are good compilations to start with) that might accept contributions.

If you would like to try to find a partner at SkyFi, we can reach out to them as well to see if they’re interested.

**Reply by edchanonline** [reply]
I am familiar with GEEMAP and its author Dr. Qiusheng Wu! He seems to be putting out great work!!

**Reply by edchanonline** [reply]
SkyFi - I may reach out to my point of contact there. It seems like they are putting together some kind of analytics plaform. Perhaps that's where I could contribute. 

![Screenshot 2024-01-25 at 4 53 30 PM](https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/4795392/56bc032c-df54-4ec6-aa64-f4487eb693a7)

**Comment by edchanonline** [comment]
**GIS Database Administration**

As some of you may know, I am from outside the earth science community. I currently work at a fintech company. I am interested in potentially exploring spatial database administration. That might be one way I could break into this field. I am not entirely sure.

**Reply by edchanonline** [reply]
University of Arizona is creating a data management and archiving plan that is responsive to SPD-41a/FAIR principals for the OSIRIS-REx program. I wonder if we could partner with someone there on a project with this data manager theme in mind.

**Reply by eculler** [reply]
Yeah, I know that data management is a big part of what a lot of folks do in EDS.

I have a couple of ideas on this front. We (ESIIL) work with U of Arizona on some cyberinfrastructure initiatives, including working on issues of data sovereignty and CARE principles, and it may be that they have a potential project. They also run a [Foundational Open Science Skills course](https://cyverse.org/foss) each year that might be of interest.

**Reply by edchanonline** [reply]
The point of contact for OSIRIS-REx Sample Analysis Micro-Information System is a data manager named Carina Bennett at University of Arizona.

**Comment by intiluna** [comment]
# Topic 1: Crop Yield Estimation or Agriculture / Soil  related

I am starting a PhD and most of my work will be around Crop Yield Estimation and Agriculture at national level for several countries. For that, a lot of variables need to be taken into account (soil properties, weather, crop type, technology, etc) and has many challenges (clouds, resolution, lack of data, crop masks and many more). In that sense, I would like to collaborate with partners working on these topics or using similar techniques or facing similar challenges that I can learn from. I have some experience and education with genetics so one topic that interested me was mentioned the first day was about soil, genetics and crop yields.

**Reply by advythr** [reply]
Cool idea! I learned recently of this resource for soils data called [SoilGrids](https://www.isric.org/explore/soilgrids). If you know of better public data sources for gridded soil data, I would love to know as that would be useful for my project also.

**Reply by sarawirth** [reply]
FAOSTAT has a lot of crop data, you could look there. Not sure if they have the kind of data you're looking for.

**Comment by peterkobylarz** [comment]
One of my major professional areas of focus is wildfire resiliency and recovery. I have worked on a number of large post wildfire projects (2018 North Bay Fires, 2018 Camp Fire, 2020 Oregon Fires, 2023 Maui Fires) and have cobbled together a number of analyses to look at big earth datasets to look at factors such as fire spread, smoke spread, etc. These tools were pretty ad-hoc and there is a lot of room for improvement. I would love to focus on this area for my main project. Some ideas include develop some Python based tools to model and animate fire and smoke spread using data from sensors such as MODIS, GOES, Sentinel 2, and others.

Other major interests include:

- Land cover classification at as high resolution as possible. In my experience regional land cover datasets are inaccurate, out-dated, and low resolution to the point where they aren't useful in a lot of environmental project applications.
- Wildfire fuel mapping and classification at high resolutions - same as above. I am very interested in developing tools to perform project/local-level fuel mapping based using LIDAR and imagery data.

**Reply by nbol48** [reply]
Peter, I have a background in wildfire and forestry and am also interested in a related project. Open to any and all ideas and the outlines you listed here seem like they could lead to great projects.

**Reply by peterkobylarz** [reply]
Right on, I'd love to brainstorm at some point!

**Reply by eculler** [reply]
I will ask, but I think this might be a good fit for working with Lilly, especially if you are looking to shape the project goals a lot. Keep posting if you two brainstorm and come up with something you want to do!

**Comment by eculler** [comment]
# Potential wildfire project with Earthlab scientist Lise St. Denis

## Wildfire data
Projects would use a dataset mined from the public archive (1999–2020) of the US National Incident Management System Incident Status Summary (ICS-209) forms (a total of 187,160 reports for 35,170 incidents, including 34,478 wildland fires). This system captures detailed **daily/regular information on (mostly wildfire) incident development and response, including social and economic impacts**. An integrated and spatially augmented dataset enables exploration of the daily progression of the most costly, damaging, and deadly environmental-hazard events in recent US history.

## Why should you care?
This research dataset is valuable, particularly for wildfire, because it captures daily snapshots of wildfire characteristics & drivers, current response (e.g. resources, outlook), impacts (e.g. structures threatened, damaged destroyed, values at risk).

## Potential projects
- Work with the USFS Human Dimensions Program to enhance the dataset including expanding the resource allocation data broken down by agency and resource type and build a topic modeling framework to support efficient mining of the narrative text contained on these reports.
- Overlay fire perimeter data with a housing layer to provide some granularity to where the impacts occurred.
- Link this data to other key data such as air quality. We are open opportunities based on student interest here.
- Look at filtered social media related to large, impactful wildfires. The social media classifier needs to be tailored to wildfire. The data could then be used to build strategies for analyzing public/community response for over 100 historic wildfires.

**Reply by nbol48** [reply]
Elsa, I am curious about the effectiveness of different fire agencies in fire suppression. I am not sure if there are enough incidents to analyze this question holding the many variables constant, but many incidents are split by division, and each division is commanded by a different agency which could provides a close comparison, not to mention the many small fires which are single agency responses. Let me know if you think this could be a worthwhile conversation.

**Reply by lucap1211** [reply]
I was already interested in this project before Lise's presentation, but after learning more about the opportunity to work with a variety of data sources, I got even more excited. I am particularly interested in the incidence response and impacts aspect of the datasets that she is working with and hope to be involved in her research for my class project. I look forward to the possibility of combining incident reporting with social media to understand public perception of wildfire risk better. Additionally, considering the wildfire perimeter data overlayed with the housing layer, it could be valuable to combine those layers with a road network layer to "score/rate" evacuation potential and compare it to data (if available) from the 100 historic fires.

**Comment by eculler** [comment]
# Potential project with The Watershed Center
[The Watershed Center](https://watershed.center/our-work/) is a community restoration and management organization in Boulder County. They are interested in developing a tool to efficiently analyze forest structure and spatial heterogeneity.

Earthlab will be hosting a forest carbon intensive workshop in mid-February, and we would be able to include 1-2 students working on this forest project if there’s interest.

**Reply by nbol48** [reply]
This concept is also very interesting to me in developing high resolution analysis of fuel loading areas of priorities for fire mitigation efforts and other forest management goals (eg. timber harvest).

**Comment by eculler** [comment]
# Potential project with NEON
The [National Ecological Observatory Network](https://www.neonscience.org/) collects and manages data for a variety of key sites across the U.S. They are interested in building curriculum to show off their data and help scientists get started using it. NEON data is fabulous because it’s an opportunity to work with really high-quality, high-resolution data collected in situ or using an airborne platform like a plane or drone. That data can then be linked with some satellite data sources to understand variation across a huge range of scales.

## Potential project ideas

1. We (NEON) have been talking to some folks at NASA about aligning training resources around NASA EMIT, AVIRIS, and NEON AOP data. Some sort of scaling application between these airborne / satellite remote sensing datasets would be great. This could have a deliverable of a tutorial or a reproducible workflow using at least two of these datasets, to be published on our website or submitted to our https://www.neonscience.org/resources/code-hub. This material might also be presented at a workshop at some time in the future. We (ESIIL) might have opportunities for the project group to help out at a workshop in June about hyperspectral and thermal data to build skills for this project.
2. Complete some exploratory analysis using Google Earth Engine and publish it as a tutorial — for example on hazard identification. There’s a lot of NEON data on GEE in addition to many satellite datasets.
3. Work on a Python package for downloading an exploring NEON data (that would duplicate the functionality of the R neonUtilities package). For example, adding tools for working with lidar point cloud data, or hyperspectral data.
4. Updating older tutorials to use the most up-to-date libraries and clean up the code. This would probably be the first step to a more open-ended project to help get familiar with the data.

**Reply by edchanonline** [reply]
I'd love to learn more about this!

**Comment by eculler** [comment]
# Potential project with Lilly

For someone who wants to do something related to wildfire, water, and/or Tribal lands, Lilly has offered to mentor a project. This is a great option if you're really into defining what your own project will be from the ground up! She's done research on hydrologic modeling for the Pine Ridge reservation, and now works on using ML, AI, and remote sensing to understand wildfire in the Western US. This would also be a great project for someone who wants to get involved in Tribal research collaborations.

**Comment by advythr** [comment]
## Potential project ideas - Advyth

For my final project, since I'm a CU grad student, I hope to start working on a remote sensing chapter for my dissertation. My project mentor will be my adviser Dr. Katharine Suding.

### Idea #1
Our ability to understand the drivers of and predict ecosystem productivity is critical to parameterizing Earth System Models and understanding feedbacks between climate and carbon cycling. Specifically, understanding which ecosystems are most sensitive to climatic variation and what traits mediate sensitivity of ecosystem productivity to climate is important to advancing our understanding. For example, [Felton et al. (2021)](https://nph.onlinelibrary.wiley.com/doi/full/10.1111/nph.17543) used Landsat-derived annual net primary productivity to understand if vegetation structure, ecoregion types, and mean climate influenced temporal sensitivity of NPP to precipitation.

I'm potentially interested in perhaps extending this line of inquiry to test what plant traits (e.g., root length, leaf area) and environmental variables (e.g., soil properties, topography) mediate ecosystem-scale productivity sensitivity to precipitation. This could involve Landsat data, soil data from [SoilGrids](https://www.isric.org/explore/soilgrids), digital elevation models and climate data.

### Idea #2
Understanding how plant traits influence ecosystem-scale flammability and thus fire properties is important for advancing fire models and identifying areas of high fire risk. For example, [Rao et al. (2022)](https://www.nature.com/articles/s41559-021-01654-2) examined what traits mediate plant sensitivity to vapor pressure deficit to identify areas of high fire hazard.

I'm potentially interested in combining [Forest Inventory & Analysis (FIA)](https://www.fs.usda.gov/research/programs/fia) data with the plant trait database [TRY](https://www.try-db.org/TryWeb/Home.php) to scale up plant traits to the ecosystem-level and then testing whether wildfire probability and multiple dimensions of fire regimes can be explained by plant traits.

**Comment by sarawirth** [comment]
I'm interested in doing a project involving ecosystems or biodiversity. I'm particularly interested in keystone and invasive species.
